src/main/java/org/springframework/data/redis/cache/DefaultRedisCacheWriter.java

```diff
@@ -179,12 +183,31 @@ public void clean(String name, byte[] pattern) {
 					wasLocked = true;
 				}
 
-				byte[][] keys = Optional.ofNullable(connection.keys(pattern)).orElse(Collections.emptySet())
-						.toArray(new byte[0][]);
+//				byte[][] keys = Optional.ofNullable(connection.keys(pattern)).orElse(Collections.emptySet())
+//					.toArray(new byte[0][]);
+//
+//				if (keys.length > 0) {
+//					connection.del(keys);
+//				}
 
-				if (keys.length > 0) {
-					connection.del(keys);
+				int batchSize = 1000; // maybe make it configurable
+				ScanOptions options = ScanOptions.scanOptions().match(new String(pattern)).build();
```

Comment:
batchSize here seems not applied to ScanOptions

Reply:
nice work
`int batchSize = 1000; // maybe make it configurable`